### ISSUE TITLE
[Bug] bug title

### ISSUE THREAD

**Issue by jacksbrain** [DLR-RM/stable-baselines3#666]
### 🐛 Bug

env.reset() does not reset an environment for Atari

### To Reproduce

I did it in Jupyter

```python
from stable_baselines3.common.env_util import make_atari_env
env = make_atari_env('BreakoutNoFrameskip-v4', n_envs=1, seed=0)

%matplotlib inline
from matplotlib import pyplot as plt

# Create Image 1
obs = env.reset()
plt.imshow(obs[0,:,:,0])
plt.show()

# Create Image 2
obs = env.reset()
plt.imshow(obs[0,:,:,0])
plt.show()
```
Image 1
![image](https://user-images.githubusercontent.com/16499588/142233813-4cae1f60-9b71-41f7-a8c3-afca533bf097.png)

Image 2
![image](https://user-images.githubusercontent.com/16499588/142233851-c8b4a8e8-b15d-4290-ab6c-49465bb2359c.png)

### Expected behavior

Observation after env.reset() should be the same, i.d. Image 1 should be equal to Image 2. But in reality Image 2 is the next step in the environment.

### System Info

```
OS: macOS-12.0.1-x86_64-i386-64bit Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64
Python: 3.8.2
Stable-Baselines3: 1.3.0
PyTorch: 1.10.0
GPU Enabled: False
Numpy: 1.21.4
Gym: 0.19.0
```

**Comment by araffin** [comment]
> But if I run your code with terminal_on_life_loss=True one can see that env.reset() is actually perform env.step(). This behaviour looks odd.

I agree and at the same time, the env is not made to be reset in cascade like that. But we should document anyway and maybe change the default as discussed in https://github.com/DLR-RM/stable-baselines3/issues/635

**Comment by IAmTheNail** [comment]
I also would go for the recommended change in #635. Because otherwise the default evaluation mechanism is broken. It's bound to once in while select under-performing models. However, I can imagine there are cases in which you'd rather continue than reset. For environments where resetting has no impact on the final result. You'd rather not spend time on useless resets.

Another idea would be to introduce the notion of a hard reset; bypassing all optimizations and clever tricks. EvalCallback would invoke it each time it starts evaluating. But that would require some refactoring.

Anyway, many thanks for the quick response and the awesome work ! :-)
